### PR TITLE
skeleton landing page (LAND-11)

### PR DESCRIPTION
**Pull request by lucassshanks**
PR to create initial landing page. Lacks a lot of copy/images still and needs some refactoring to be used in mobile. also unsure if we want to put stuff in different places. also lacks a bit of standardization in how some components are made. Think it should all just be scss and maybe remove the Flex component since with scss it gets a little weird and turns out i don't mind using it that much

## Motivation

Adding the initial scaffolding for the landing page.

[Figma](https://www.figma.com/design/7Edv1qWpuVIjKTFMG1tt3F/Drafts?node-id=274-1755&node-type=canvas&t=x1rmIV3H8xAo8kNm-0)

## Solution

-.-

## Tickets

LAND-11

### CODE REVIEW

**Comment by TroyKomodo** [comment]
We should definitely setup prettier or some formatter / linter.

File: package.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{
+  "name": "test",
```
Comment (TroyKomodo) [inline package.json]: Lets call this `@scufflecloud/landing`

File: package.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  "version": "0.0.1",
+  "type": "module",
+  "scripts": {
+    "dev": "vite dev",
+    "build": "vite build",
+    "preview": "vite preview",
+    "check": "svelte-kit sync && svelte-check --tsconfig ./tsconfig.json",
+    "check:watch": "svelte-kit sync && svelte-check --tsconfig ./tsconfig.json --watch",
+    "format": "prettier --write ."
```
Comment (TroyKomodo) [inline package.json]: Do we have a script to run eslint or to check for prettier issues? is that the same as `check`?

File: src/app.d.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+// See https://svelte.dev/docs/kit/types#app.d.ts
```
Comment (TroyKomodo) [inline src/app.d.ts]: Is this file needed?

File: src/app.css [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+h2 {
+  font-size: 2.5rem;
+}
+
+h3 {
+  font-size: 1.5rem;
+}
+
+body {
+  min-height: 100vh;
```
Comment (TroyKomodo) [inline src/app.css]: Do we have a custom body scrollbar?

File: src/lib/components/Homepage/EmailForm.svelte [hunk 5]
```diff
@@ -0,0 +1,82 @@
+<script lang="ts">
+  import Pill from '$lib/design-components/Pill.svelte';
+  import { theme } from '$lib/theme';
+</script>
+
+<div class="rectangle">
+  <div class="update-banner">
+    <div style="padding: 0px 8px;">
```
Comment (TroyKomodo) [inline src/lib/components/Homepage/EmailForm.svelte]: Can this be a class? and not `px` rather `rem` or `em`

File: src/lib/components/GithubStats.svelte [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+  }
+  :global(.pill-link > *) {
+    height: 100%;
+  }
+
+  .divider {
+    width: 2px;
+    height: 20px;
+    background-color: #ddd;
+    border-radius: 1px;
```
Comment (TroyKomodo) [inline src/lib/components/GithubStats.svelte]: `px` units instead of relative

File: src/lib/components/Homepage/EmailForm.svelte [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+    <div class="entry-field-container">
+      <input class="entry-field" placeholder="Your email address" type="email" />
+      <Pill color={theme.colors.brown600}>
+        <p class="button-text">Subscribe</p>
+      </Pill>
+    </div>
+  </div>
+</div>
+
+<style lang="scss">
```
Comment (TroyKomodo) [inline src/lib/components/Homepage/EmailForm.svelte]: I think this doesnt do anything?

File: src/lib/components/Navbar.svelte [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+    max-width: 64rem;
+    margin: 0 auto;
+    padding: 1rem;
+    width: 100%;
+  }
+
+  header {
+    display: flex;
+    justify-content: space-between;
+    max-width: 1000px;
```
Comment (TroyKomodo) [inline src/lib/components/Navbar.svelte]: Where does 1000px come from?

File: src/lib/components/Navbar.svelte [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+  ul {
+    position: relative;
+    padding: 0;
+    margin: 0;
+    height: 3em;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    list-style: none;
+    background: var(--background);
```
Comment (TroyKomodo) [inline src/lib/components/Navbar.svelte]: Where is this defined?

File: svelte.config.js [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+    // If your environment is not supported, or you settled on a specific environment, switch out the adapter.
+    // See https://svelte.dev/docs/kit/adapters for more information about adapters.
+    adapter: adapter(),
+  },
+  vitePlugin: {
+    dynamicCompileOptions({ filename }) {
+      if (filename.includes('node_modules')) {
+        return { runes: undefined };
+      }
+    },
```
Comment (TroyKomodo) [inline svelte.config.js:22]: What does this do?

File: vite.config.ts [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+
+export default defineConfig({
+  plugins: [sveltekit()],
+  // https://stackoverflow.com/questions/78997907/the-legacy-js-api-is-deprecated-and-will-be-removed-in-dart-sass-2-0-0
+  css: {
+    preprocessorOptions: {
+      scss: {
+        api: 'modern-compiler',
+      },
+    },
```
Comment (TroyKomodo) [inline vite.config.ts]: I dont think we need scss, maybe?

**Review by TroyKomodo — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (lucassshanks) [reply package.json]: Screenshot 2024-12-04 at 9 33 19 PM
yeah pnpm format runs the formatter. check will tell u if u have errors

File: src/app.css (hunk 4 above)
Comment (lucassshanks) [reply src/app.css]: no. should i add one

File: package.json (hunk 2 above)
Comment (TroyKomodo) [reply package.json]: im saying does this `pnpm check` does this run eslint as part of svelte-check or do u still have to run eslint?

File: src/lib/components/Homepage/EmailForm.svelte (hunk 7 above)
Comment (lucassshanks) [reply src/lib/components/Homepage/EmailForm.svelte]: i guess you're right.. my syntax highlighting looks a little odd sometimes without it but safe to remove

File: src/lib/components/Navbar.svelte (hunk 8 above)
Comment (lucassshanks) [reply src/lib/components/Navbar.svelte]: removed

File: src/lib/components/Navbar.svelte (hunk 9 above)
Comment (lucassshanks) [reply src/lib/components/Navbar.svelte]: wasn't doing anything 👍

File: svelte.config.js (hunk 10 above)
Comment (lucassshanks) [reply svelte.config.js:22]: left comment

File: vite.config.ts (hunk 11 above)
Comment (lucassshanks) [reply vite.config.ts]: uhhh. yeah ig ill just remove it. i initially thought nesting required it but i suppose i dont need it then

File: src/app.css (hunk 4 above)
Comment (lucassshanks) [reply src/app.css]: will do in follow up pr

File: src/app.css [hunk 12]
```diff
@@ -0,0 +1,86 @@
...
+:root {
+  --font-body: Arial, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu,
+    Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
+  --font-mono: 'Fira Mono', monospace;
+
+  font-size: 16px;
+  --color-text: rgba(0, 0, 0, 0.7);
+  --column-width: 42rem;
+  --column-margin-top: 4rem;
+  --breakpoint-md: 48rem;
```
Comment (TroyKomodo) [inline src/app.css]: Did we figure out how to do this?

File: src/lib/components/GithubStats.svelte [hunk 13]
```diff
@@ -0,0 +1,51 @@
+<script lang="ts">
+  import Flex from '$lib/design-components/Flex.svelte';
+  import Pill from '$lib/design-components/Pill.svelte';
+  import github from '$lib/images/github.svg';
+  const REPO_ID = 594510762;
```
Comment (TroyKomodo) [inline src/lib/components/GithubStats.svelte]: can this come from a vite env?

File: src/lib/components/GithubStats.svelte [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+  }
+  :global(.pill-link > *) {
+    height: 100%;
+  }
+
+  .divider {
+    width: 2px;
+    height: 1.25rem;
+    background-color: #ddd;
+    border-radius: 1px;
```
Comment (TroyKomodo) [inline src/lib/components/GithubStats.svelte]: i want you to disable the letters `px` on your keyboard.

File: src/lib/components/Homepage/EmailForm.svelte [hunk 15]
```diff
@@ -0,0 +1,86 @@
...
+  }
+
+  .entry-field {
+    padding: 0.75rem 1rem;
+    border-radius: 100px;
+    border: 1px solid #ffdbb9;
+    width: 100%;
+    outline: none;
+    transition: all 0.2s ease;
+    background-color: inherit;
```
Comment (TroyKomodo) [inline src/lib/components/Homepage/EmailForm.svelte]: px

File: src/lib/components/Homepage/Footer.svelte [hunk 16]
```diff
@@ -0,0 +1,81 @@
...
+      gap: 0.75rem;
+
+      a {
+        font-size: 0.875rem;
+      }
+    }
+  }
+
+  /* Can't define breakpoint variables without scss  */
+  @media (max-width: 768px) {
```
Comment (TroyKomodo) [inline src/lib/components/Homepage/Footer.svelte:75]: this is fine, can you just say why 768px is the right value where did u get it from?

File: src/lib/components/Navbar.svelte [hunk 17]
```diff
@@ -0,0 +1,115 @@
...
+
+  /* Active page indicator */
+  li[aria-current='page']::before {
+    content: '';
+    position: absolute;
+    bottom: 0;
+    left: 50%; /* Center the line */
+    transform: translateX(-50%); /* Offset by half its width for true centering */
+    width: 60%; /* Adjust this value to make it wider/narrower */
+    height: 3px;
```
Comment (TroyKomodo) [inline src/lib/components/Navbar.svelte]: px

File: src/lib/design-components/Pill.svelte [hunk 18]
```diff
@@ -0,0 +1,35 @@
...
+>
+  {@render children()}
+</div>
+
+<style>
+  .pill {
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    padding: 0px 1rem;
```
Comment (TroyKomodo) [inline src/lib/design-components/Pill.svelte]: `padding: 0 1rem`

File: src/lib/design-components/Pill.svelte [hunk 19]
```diff
@@ -0,0 +1,35 @@
...
+    border-radius: 50rem;
+    width: fit-content;
+    cursor: pointer;
+    background-color: var(--pill-background);
+    border: 1px solid var(--pill-border-color);
```
Comment (TroyKomodo) [inline src/lib/design-components/Pill.svelte:49]: px

File: src/lib/components/Navbar.svelte [hunk 20]
```diff
@@ -0,0 +1,115 @@
...
+    display: flex;
+    justify-content: space-between;
+    width: 100%;
+    margin: 0 auto;
+  }
+
+  .corner {
+    height: 3em;
+    display: flex;
+    gap: 0.5rem;
```
Comment (TroyKomodo) [inline src/lib/components/Navbar.svelte]: is there a specific reason you choose to us `em` and then `rem` does this make sense?

File: src/lib/index.ts [hunk 21]
```diff
@@ -0,0 +1 @@
+// place files you want to import through the `$lib` alias in this folder.
```
Comment (TroyKomodo) [inline src/lib/index.ts]: this file can be deleted i think?

File: src/routes/+page.svelte [hunk 22]
```diff
@@ -0,0 +1,444 @@
...
+    }
+  }
+
+  /* Will be changed later when picture is added */
+  .box {
+    width: 100px;
+    height: 100px;
+    background: var(--color-light200);
+    border-radius: 0.5rem;
+  }
```
Comment (TroyKomodo) [inline src/routes/+page.svelte:239]: `px` im not apposed to leaving this since the comment says you will change it but; i suspect we might forget to do that

File: src/routes/+page.svelte [hunk 23]
```diff
@@ -0,0 +1,444 @@
...
+  .solution-title {
+    font-weight: 400;
+    margin-bottom: 2rem;
+    font-size: 4rem;
+  }
+
+  .feature-description {
+    font-size: 1rem;
+    opacity: 0.8;
+    max-width: 600px;
```
Comment (TroyKomodo) [inline src/routes/+page.svelte]: px, also why 600px

File: src/routes/+page.svelte [hunk 24]
```diff
@@ -0,0 +1,444 @@
...
+    line-height: 1.5;
+  }
+
+  .features-performance {
+    h2 {
+      font-size: 2.5rem;
+      margin-bottom: 3rem;
+      @media (max-width: 768px) {
```
Comment (TroyKomodo) [inline src/routes/+page.svelte:282]: what does 768px mean?

File: src/routes/+page.svelte [hunk 25]
```diff
@@ -0,0 +1,444 @@
...
+    }
+  }
+
+  .features-pill {
+    display: inline-flex;
+    align-items: center;
+    justify-content: center;
+    padding: 0.5rem 1rem;
+    border: 1px solid var(--color-orange500);
+    border-radius: 8px;
```
Comment (TroyKomodo) [inline src/routes/+page.svelte]: px

Comment (TroyKomodo) [reply src/routes/+page.svelte]: all other references in this file

File: src/app.css (hunk 12 above)
Comment (lucassshanks) [reply src/app.css]: cleaned up

File: vite.config.ts (hunk 11 above)
Comment (TroyKomodo) [reply vite.config.ts]: you resolved this but didnt remove it

**Review by TroyKomodo — CHANGES_REQUESTED** [review]

File: src/lib/design-components/Pill.svelte (hunk 19 above)
Comment (lucassshanks) [reply src/lib/design-components/Pill.svelte:49]: as discussed this is fine so marking as resolved

File: src/routes/+page.svelte (hunk 22 above)
Comment (lucassshanks) [reply src/routes/+page.svelte:239]: i won't forget ill remove this whole box

File: src/routes/+page.svelte (hunk 24 above)
Comment (lucassshanks) [reply src/routes/+page.svelte:282]: left comment

**Review by TroyKomodo — APPROVED** [review]
